features/steps/notification_database.py

```diff
@@ -264,3 +260,34 @@ def insert_report_with_risk_and_cooldown_in_new_reports_table(context, risk):
     """Insert rows into table new_reports after the cooldown has passed."""
     timestamp_after_cooldown = datetime.now() + timedelta(minutes=1)
     insert_report_with_risk_in_new_reports_table(context, risk, updated_at=timestamp_after_cooldown)
+
+
+@given("I insert 1 previously reported report with {risk:w} total risk")
+def insert_report_into_reported_table(context, risk, timestamp=None):
+    """Inserts rows into reported table."""
+    report = generate_report_with_risk(risk)
+    insert_rows_into_reported_table(context, report, timestamp)
+
+
+@given("I insert 1 previously reported report with {risk:w} total risk notified within cooldown")
+def insert_report_within_cooldown_in_reported_table(context, risk):
+    """Inserts rows into reported table within cooldown."""
+    timestamp_within_cooldown = datetime.now() - timedelta(seconds=30)
+    insert_report_into_reported_table(context, risk, timestamp_within_cooldown)
+
+
+def generate_report_with_risk(risk):
+    """Creates a report with specified risk."""
+    report = '{"analysis_metadata":{"metadata":"some metadata"},"reports":[{"rule_id":"test_rule|<replace_me>","component":"ccx_rules_ocp.external.rules.test_rule.report","type":"rule","key":"<replace_me>","details":"some details"}]}'  # noqa E501
+    if risk == "critical":
```

Comment:
I'd replace it with map with risk -> replacement strings :)

```
r["critical"] = "TEST_RULE_CRITICAL_IMPACT"
```

then

```
report = report.replace("<replace me>", r[risk])
```

Reply:
Yes, that looks better. Changed.